### PR TITLE
feat: add setCookie documentation for hostOnly option

### PR DESCRIPTION
**Pull request by AtofStryker**
Adds the documentation needed for the added `hostOnly` option for `cy.setCookie`.

(NOTE: needs to be based off the targeted release branch)

### CODE REVIEW

File: docs/api/commands/setcookie.mdx [hunk 1]
```diff
@@ -38,6 +38,7 @@ Pass in an options object to change the default behavior of `cy.setCookie()`.
 | `log`      | `true`                                                         | Displays the command in the [Command log](/guides/core-concepts/cypress-app#Command-Log)                                                                                                                              |
 | `domain`   | Hostname of the current URL                                    | The domain the cookie is visible to                                                                                                                                                                                   |
 | `expiry`   | 20 years into the future                                       | When the cookie expires, specified in seconds since [Unix Epoch](https://en.wikipedia.org/wiki/Unix_time).                                                                                                            |
+| `hostOnly` | `false`                                                        | Whether the cookie is a host only cookie
```
Comment (emilyrohrbough) [inline docs/api/commands/setcookie.mdx:41]: @AtofStryker What change is this doc update related to? I has assumed https://github.com/cypress-io/cypress/pull/25761, but we didn't add this option to the command or cookie typescript types. Also, looks like a TODO still in the code: https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cy/commands/cookies.ts#L7

Comment (emilyrohrbough) [reply docs/api/commands/setcookie.mdx:41]: pulled from https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/Cookie
```suggestion
| `hostOnly` | `false`                                                        | Whether the cookie is a host-only cookie, (i.e. the request's host must exactly match the domain of the cookie)
```

Comment (AtofStryker) [reply docs/api/commands/setcookie.mdx:41]: While I was in the cookie deep dive I figured I could just add this in https://github.com/cypress-io/cypress/pull/25853 since it's pretty simple. I'll remove the todo in the PR. My guess is this change doesn't make it into the patch 😅

**Review by mschile — APPROVED** [review]
